crates/ruma-client-api/src/r0/account/add_3pid.rs

```diff
@@ -20,10 +21,10 @@ ruma_api! {
         pub auth: Option<AuthData<'a>>,
 
         /// Client-generated secret string used to protect this session.
-        pub client_secret: &'a str,
+        pub client_secret: &'a ClientSecretBox,
```

Comment:
Actually just noticed this now. Taking a borrowed `Box` is almost never the right thing to do. This should be `&'a ClientSecret` (which `&'a ClientSecretBox` can deref to).

Reply:
That's what I thought. Though for some reason I can't figure out, the compiler complain with
```
error[E0277]: the size for values of type `str` cannot be known at compilation time
  --> crates/ruma-client-api/src/r0/contact/request_contact_verification_token.rs:19:32
   |
19 |         pub client_secret: &'a ClientSecret,
   |                                ^^^^^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: within `ClientSecret`, the trait `Sized` is not implemented for `str`
   = note: required because it appears within the type `ClientSecret`
   = note: only the last field of a struct may have a dynamically sized type
   = help: change the field's type to have a statically known size
   ```
   
   I don't understand what I'm missing, because it works fine when I use `ServerName` instead.

Reply:
Right, there's some magic going on for `ServerName`, `str` and a few other types. You need to add the new types [here](https://github.com/ruma/ruma/blob/1eb2f39593b702362b5850c24cd7eb3ac913df06/crates/ruma-serde-macros/src/outgoing.rs#L261).